openhcl/virt_mshv_vtl/src/processor/mshv/x64.rs

```diff
@@ -610,9 +609,8 @@ impl<'a, 'b> InterceptHandler<'a, 'b> {
             .next_deliverability_notifications
             .set_sints(0);
 
-        // These messages are always VTL0, as VTL1 does not own any VMBUS channels.
         self.vp
-            .deliver_synic_messages(GuestVtl::Vtl0, message.deliverable_sints);
```

Comment:
Yeah you're correct. I missed the VTL earlier, so I think this change should be good.

Reply:
Or wait, was the issue here that the vtl we get is the one that sent the messages, rather than where they're targeted? If that's the case then we wouldn't want this, and I had it correct before.

Reply:
That seems plausible, but let's get confirmation. @luishmsft @benpye Can either of you help us answer this? The question is, if VTL 2 in a SNP CVM is handling a HV_SYNIC_SINT_DELIVERABLE_MESSAGE, whether the VTL field in the header of that message is the target VTL of the message or the one that happened to be running when the exit into the hypervisor happened?

Reply:
And if it is not the target VTL, does that apply to other message types too? I.e. could our other handlers above be incorrect?